core/common/src/main/java/alluxio/underfs/Fingerprint.java

```diff
@@ -30,14 +32,18 @@
   /** These tags are required in all fingerprints. */
   private static final Tag[] REQUIRED_TAGS = {Tag.TYPE, Tag.UFS, Tag.OWNER, Tag.GROUP, Tag.MODE};
   /** These tags are optional, and are serialized after the required tags. */
-  private static final Tag[] OPTIONAL_TAGS = {Tag.CONTENT_HASH};
+  private static final Tag[] OPTIONAL_TAGS = {Tag.CONTENT_HASH, Tag.ACL};
 
   /** These tags are all the metadata tags in the fingerprints. */
-  private static final Tag[] METADATA_TAGS = {Tag.OWNER, Tag.GROUP, Tag.MODE};
+  private static final Tag[] METADATA_TAGS = {Tag.OWNER, Tag.GROUP, Tag.MODE, Tag.ACL};
   /** These tags are all the content tags in the fingerprints. */
   private static final Tag[] CONTENT_TAGS = {Tag.TYPE, Tag.UFS, Tag.CONTENT_HASH};
 
-  private static final Pattern SANITIZE_REGEX = Pattern.compile("[ :]");
+  private static final char KVDELIMTER = '|';
+  private static final char TAGDELIMTER = ' ';
+
+  private static final Pattern SANITIZE_REGEX = Pattern.compile("[" + KVDELIMTER
```

Comment:
Does this mean an upgraded server will invalidate all fingerprints, regardless if it is metadata or not?

Reply:
Unfortunately yes. The ACL string will contain `:`, we would have to change the delimiter to some other symbol, which will invalidate all fingerprints.